### PR TITLE
Add recommended requirements answer to FAQ

### LINKED ISSUE #721: Recommended requirements for hosting
**Is your feature request related to a problem? Please describe.**
Recommended requirements for hosting

**Describe the solution you'd like**
Would like to see some documentation similar to this on MeiliSearch:
[https://strapi.io/documentation/developer-docs/latest/getting-started/deployment.html#recommended-requirements](https://strapi.io/documentation/developer-docs/latest/getting-started/deployment.html#recommended-requirements).

More specially, I'm looking for number of CPU cores and memory to host MeiliSearch in a production server.

### PR DESCRIPTION
**Pull request by react-learner**
Add information from #721 to the FAQ.

Closes #721

### CODE REVIEW

File: create/faq.md [hunk 1]
```diff
@@ -189,4 +176,41 @@ See our [contact page](/learn/what_is_meilisearch/contact.md).
...
+MeiliSearch is designed to be fast (≤50ms response time), so speeding it up is rarely necessary. However, if you find that your MeiliSearch instance is querying slowly, there are two primary methods to speed it up:
+
+- Increase the amount of RAM.
+- Reduce the size of your database.
+
+In general, we recommend the former. However, if you need to reduce the size of your database for any reason, keep in mind that:
+
+- **More relevancy rules => a larger database**
+  - The proximity ranking rule alone can be responsible for almost 80% of database size
+- Faceting also consumes a large amount of disk space.
```
Comment (qdequele) [inline create/faq.md:216]:
```suggestion
- Faceting also consumes a large amount of disk space.
- Split your dataset, one spoken language per index.
- Add stopwords
- Select carefully indexed attributes
    - Do not index uniques ids
```

**Review by qdequele — CHANGES_REQUESTED** [review]

File: create/faq.md [hunk 2]
```diff
@@ -189,4 +176,41 @@ See our [contact page](/learn/what_is_meilisearch/contact.md).
...
+**Short answer**: It's impossible to say. We recommend [experimenting with RAM availability and database size](#how-does-hardware-affect-search-speed) to find the hardware + relevancy setup that works best for your dataset.
+
+**Long answer**: The disk size and RAM usage of your MeiliSearch database can vary widely. The following factors have a great effect on space and memory use:
+
+- The number of documents
+- The size of documents
+- The number of indexed fields
+- The number of faceted fields
+- The size of each update
+- The number of different words present in documents (warning on multi-lingual datasets and fields with unique words like ids)
```
Comment (react-learner) [inline create/faq.md:192]:
```suggestion
- The number of different words present in documents (beware heavily multi-lingual datasets and fields with unique words, such as IDs)
```

File: create/faq.md [hunk 3]
```diff
@@ -189,4 +176,45 @@ See our [contact page](/learn/what_is_meilisearch/contact.md).
...
+
+In general, we recommend the former. However, if you need to reduce the size of your database for any reason, keep in mind that:
+
+- **More relevancy rules => a larger database**
+  - The proximity ranking rule alone can be responsible for almost 80% of database size
+- Faceting also consumes a large amount of disk space.
+- Split your dataset, one spoken language per index.
+- Add stopwords
+- Select carefully indexed attributes
+    - Do not index uniques ids
```
Comment (react-learner) [inline create/faq.md:220]:
```suggestion
- Faceting also consumes a large amount of disk space
- Multi-lingual datasets are costly, so split your dataset—one language per index
- Make sure to add [stop words](/reference/features/stop_words.md)
- Be selective about which attributes you index and make [searchable](/reference/features/field_properties.md#searchable-fields)
    - Avoid indexing unique IDs
```

File: create/faq.md [hunk 4]
```diff
@@ -189,4 +176,45 @@ See our [contact page](/learn/what_is_meilisearch/contact.md).
 
 ## I have just updated MeiliSearch, and I am getting an error: "Cannot open database, expected MeiliSearch engine version..."
 
-MeiliSearch minor versions are not compatible with each other because the way we represent data internally changes with the addition of new breaking features. To fix this, it suffices to delete your database folder (`data.ms` by default) and re-index your documents with the new version.
+MeiliSearch minor versions are not compatible with each other because the way we represent data internally changes with the addition of new breaking features. To fix this, it suffices to delete your database folder (`data.ms` by default) and re-index your documents with the new version. See [updating MeiliSearch](/reference/features/installation.md#updating-meilisearch) for more information.
```
Comment (fharper) [inline create/faq.md:179]: are = may?

Also, we should specify it's for now, or until v1 where only major version may bring breaking changes.

File: create/faq.md [hunk 5]
```diff
@@ -189,4 +176,45 @@ See our [contact page](/learn/what_is_meilisearch/contact.md).
...
+
+## What are the recommended requirements for hosting a MeiliSearch instance?
+
+**Short answer**: It's impossible to say. We recommend [experimenting with RAM availability and database size](#how-does-hardware-affect-search-speed) to find the hardware + relevancy setup that works best for your dataset.
+
+**Long answer**: The disk size and RAM usage of your MeiliSearch database can vary widely. The following factors have a great effect on space and memory use:
```
Comment (fharper) [inline create/faq.md:185]: Are they listed in DESC order of impact? If so, we should say it. If not, we should list them in order if possible. If not possible, we should mention it.

File: create/faq.md [hunk 6]
```diff
@@ -189,4 +176,45 @@ See our [contact page](/learn/what_is_meilisearch/contact.md).
...
+All of this means that **it's almost impossible to estimate the size and memory usage of a MeiliSearch database** before creation, even based on the size of its documents. There are, however, a few [general principles](#how-does-hardware-affect-search-speed) you should know.
+
+## How does hardware affect search speed?
+
+Because MeiliSearch uses a [memory map](/reference/under_the_hood/storage.md#lmdb), **search speed is based on the ratio between RAM and database size**. In other words:
+
+- A big database + a small amount of RAM => slow search
+- A small database + tons of RAM => lightning fast search
+
+The number of CPU cores has no impact on index speed (indexing is single-threaded... for now 😉). However, **the more cores you provide to the engine, the more search queries it will be able to process at the same time**.
```
Comment (fharper) [inline create/faq.md:203]: What does for now means? Will we use multi-core threading for indexing with the new engine. If so, we can say it now.

Comment (fharper) [inline create/faq.md:203]: no impact on index speed and query speed?

File: create/faq.md [hunk 7]
```diff
@@ -189,4 +176,45 @@ See our [contact page](/learn/what_is_meilisearch/contact.md).
...
+
+### Speeding Up MeiliSearch
+
+MeiliSearch is designed to be fast (≤50ms response time), so speeding it up is rarely necessary. However, if you find that your MeiliSearch instance is querying slowly, there are two primary methods to speed it up:
```
Comment (fharper) [inline create/faq.md:207]: It sounds marketing, but `≤50ms response time` means nothing. Is it always that fast no matter the number of documents or the type of queries? Maybe we should be more precise on this stats, don't want it to fuck us up in the future :)

**Review by fharper — CHANGES_REQUESTED** [review]

File: create/faq.md (hunk 5 above)
Comment (react-learner) [reply create/faq.md:185]: Good point. I'll ask @qdequele about it in the next docs sync meeting. For now, I think we should forge ahead and merge this PR.

File: create/faq.md (hunk 7 above)
Comment (react-learner) [reply create/faq.md:207]: Apparently the 50ms number actually has some basis in the code. Will check w/ core for more details :)

**Review by fharper — APPROVED** [review]
JGTM

File: create/faq.md [hunk 8]
```diff
@@ -189,4 +176,50 @@ See our [contact page](/learn/what_is_meilisearch/contact.md).
...
+**Short answer**: It's impossible to say. We recommend [experimenting with RAM availability and database size](#how-does-hardware-affect-search-speed) to find the hardware + relevancy setup that works best for your dataset.
+
+**Long answer**: The disk size and RAM usage of your MeiliSearch database can vary widely. The following factors have a great effect on space and memory use:
+
+- The number of documents
+- The size of documents
+- The number of indexed fields
+- The number of faceted fields
+- The size of each update
+- The number of different words present in documents (beware heavily multi-lingual datasets and fields with unique words, such as IDs)
```
Comment (qdequele) [inline create/faq.md:192]:
```suggestion
- The number of different words present in documents (beware heavily multi-lingual datasets and indexed fields with unique words, such as IDs or URLs)
```

Comment (qdequele) [reply create/faq.md:192]: It's not a necessary change until you already talk about that on the last line of the page.

**Review by qdequele — APPROVED** [review]

**Pull request merged**